runtime/compiler/control/JITServerCompilationThread.cpp

```diff
@@ -664,28 +687,34 @@ TR::CompilationInfoPerThreadRemote::processEntry(TR_MethodToBeCompiled &entry, J
       // If the client aborted this compilation it could have happened only while asking for entire
       // CHTable and unloaded class address ranges and at that point the seqNo was not updated.
       // We must update seqNo now to allow for blocking threads to pass through.
+      // Since the caches are already cleared there is no harm in discarding this message.
       clientSession->getSequencingMonitor()->enter();
 
-      if (seqNo != clientSession->getExpectedSeqNo())
+     // if (isCriticalRequest) TODO: what should we do here?
          {
-         Trc_JITServerUnexpectedSeqNo(compThread, getCompThreadId(), clientSession,
-               (unsigned long long)clientId, seqNo, clientSession->getExpectedSeqNo(), clientSession->getNumActiveThreads());
+         if (seqNo > clientSession->getLastProcessedCriticalSeqNo())
+            {
+           
+            clientSession->setLastProcessedCriticalSeqNo(seqNo);
+      
+     
+            //Trc_JITServerUnexpectedSeqNo(compThread, getCompThreadId(), clientSession,
+            //   (unsigned long long)clientId, seqNo, clientSession->getExpectedSeqNo(), clientSession->getNumActiveThreads());
 
-         TR_ASSERT(false, "compThreadID=%d clientSessionData=%p clientUID=%llu (seqNo=%u, expectedSeqNo=%u, numActiveThreads=%d) unexpected seqNo",
-               getCompThreadId(), clientSession, (unsigned long long)clientId,
-               seqNo, clientSession->getExpectedSeqNo(), clientSession->getNumActiveThreads());
+            //TR_ASSERT(false, "compThreadID=%d clientSessionData=%p clientUID=%llu (seqNo=%u, expectedSeqNo=%u, numActiveThreads=%d) unexpected seqNo",
+            //   getCompThreadId(), clientSession, (unsigned long long)clientId,
+            //   seqNo, clientSession->getExpectedSeqNo(), clientSession->getNumActiveThreads());
 
-         clientSession->setExpectedSeqNo(seqNo);
+            }
          }
 
-      updateSeqNo(clientSession);
+      //updateSeqNo(clientSession);
```

Comment:
Does this TODO still need to be addressed? If so, can we put it in an issue with some description of what needs to be done to clean it up? If not, let's just delete this code.

Reply:
I changed the code to address the TODO